File: Shared/Models/ServiceAlert.swift
import Foundation

// Alerta de servicio. Viaja embebida en /infraestructura/gerencias y /infraestructura/ramales.
struct ServiceAlert: Identifiable, Hashable {
    let id: Int
    let lineId: Int?
    let ramalId: Int?
    let causeGTFS: String
    let effectGTFS: String
    let content: String
    let criticality: Int
    let validFrom: Date?
    let validUntil: Date?

    // Mapea causa/efecto GTFS a SF Symbols. El efecto manda sobre la causa.
    var iconSystemName: String {
        switch effectGTFS {
        case "NO_SERVICE":         return "xmark.octagon.fill"
        case "SIGNIFICANT_DELAYS": return "clock.badge.exclamationmark.fill"
        case "REDUCED_SERVICE":    return "tram.fill.tunnel"
        case "DETOUR":             return "arrow.triangle.branch"
        case "STOP_MOVED":         return "mappin.slash"
        default: break
        }
        switch causeGTFS {
        case "ACCIDENT":          return "exclamationmark.triangle.fill"
        case "TECHNICAL_PROBLEM": return "wrench.and.screwdriver.fill"
        case "STRIKE":            return "person.2.slash.fill"
        case "MAINTENANCE":       return "hammer.fill"
        case "WEATHER":           return "cloud.bolt.rain.fill"
        default:                  return "exclamationmark.circle.fill"
        }
    }
}

File: android/app/src/main/java/com/alandaitch/anden/data/model/ServiceAlert.kt
package com.alandaitch.anden.data.model

import java.time.Instant

// Alerta de servicio. Viaja embebida en /infraestructura/gerencias y /infraestructura/ramales.
data class ServiceAlert(
    val id: Int,
    val lineId: Int?,
    val ramalId: Int?,
    val causeGTFS: String,
    val effectGTFS: String,
    val content: String,
    val criticality: Int,
    val validFrom: Instant?,
    val validUntil: Instant?
) {
    // Nombre de ícono Material. El efecto manda sobre la causa.
    // Los agentes de UI mapean estos nombres a íconos concretos.
    val iconName: String
        get() {
            when (effectGTFS) {
                "NO_SERVICE" -> return "block"
                "SIGNIFICANT_DELAYS" -> return "schedule_alert"
                "REDUCED_SERVICE" -> return "train_tunnel"
                "DETOUR" -> return "alt_route"
                "STOP_MOVED" -> return "wrong_location"
            }
            return when (causeGTFS) {
                "ACCIDENT" -> "warning"
                "TECHNICAL_PROBLEM" -> "build"
                "STRIKE" -> "person_off"
                "MAINTENANCE" -> "handyman"
                "WEATHER" -> "thunderstorm"
                else -> "error"
            }
        }
}
